test/libsolidity/semanticTests/storage/storage_boundary_struct_array_packed.sol

```diff
@@ -0,0 +1,125 @@
+contract C {
+    struct S {
+        uint64 a;
+        uint64 b;
+        uint64 c;
+        uint64 d;
+        // All fit in one slot (4 * 64 = 256 bits)
+    }
+
+    struct Neighbor {
+        uint256 value;
+    }
+
+    function getBoundaryArray() internal pure returns (S[10][1] storage arr) {
+        // 10 structs, each 1 slot = 10 slots total
+        assembly {
+            arr.slot := sub(0, 5)
+        }
+    }
+
+    function getDest() internal pure returns (S[10][1] storage arr) {
+        assembly {
+            arr.slot := 6
+        }
+    }
+
+    function getNeighbor() internal pure returns (Neighbor storage n) {
+        // Array ends at slot 4, neighbor at slot 5
+        assembly {
+            n.slot := 5
+        }
+    }
+
+    constructor() {
+        getNeighbor().value = 0x42424242;
```

Comment:
I'd fill the whole slot with ones so that clearing any part of it can be noticed.
```suggestion
        getNeighbor().value = type(uint).max;
```